### PR TITLE
Add new versions for log4cplus.

### PR DESCRIPTION
**Pull request by weijianwen**

### CODE REVIEW

File: var/spack/repos/builtin/packages/log4cplus/package.py [hunk 1]
```diff
@@ -31,6 +31,9 @@ class Log4cplus(CMakePackage):
     granular control over log management and configuration."""
 
     homepage = "https://sourceforge.net/projects/log4cplus/"
-    url      = "https://nchc.dl.sourceforge.net/project/log4cplus/log4cplus-stable/1.2.0/log4cplus-1.2.0.tar.gz"
+    url      = "https://download.sourceforge.net/project/log4cplus/log4cplus-stable/2.0.1/log4cplus-2.0.1.tar.bz2"
 
+    version('2.0.1', 'ec01c03241ebd31127a44d1880830d8f')
+    version('1.2.0', '7f98418ab25273a5b44cb377b54cbbfb')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/log4cplus/package.py]: there are two `1.2.0` now.

**Comment by weijianwen** [comment]
@davydden fixed.

File: var/spack/repos/builtin/packages/log4cplus/package.py [hunk 2]
```diff
@@ -31,6 +31,12 @@ class Log4cplus(CMakePackage):
     granular control over log management and configuration."""
 
     homepage = "https://sourceforge.net/projects/log4cplus/"
-    url      = "https://nchc.dl.sourceforge.net/project/log4cplus/log4cplus-stable/1.2.0/log4cplus-1.2.0.tar.gz"
+    url      = "https://download.sourceforge.net/project/log4cplus/log4cplus-stable/2.0.1/log4cplus-2.0.1.tar.bz2"
 
+    def url_for_version(self, version):
+        url = 'https://download.sourceforge.net/project/log4cplus/log4cplus-stable/{0}/log4cplus-{1}.tar.bz2'
+        return url.format(version.up_to(3), version)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/log4cplus/package.py]: If you aren't going to add 1.0.4.3 anymore, you don't actually need this `url_for_version`. If you are going to add it, I like to put the `url_for_version` below the version()` directives.

Comment (weijianwen) [reply var/spack/repos/builtin/packages/log4cplus/package.py]: Version `1.0.4.3` seems like a buggy version which has never been built successfully. I've removed this version along with `url_for_version`.

**Pull request merged**